mungegithub/mungers/bulk-lgtm.go

```diff
@@ -111,15 +112,17 @@ func (b *BulkLGTM) Munge(obj *github.MungeObject) {
 	b.currentPRList[*pr.Number] = obj
 }
 
-// RegisterOptions registers config options for this munger.
-func (b *BulkLGTM) RegisterOptions(opts *options.Options) {
+// RegisterOptions registers config options for this munger, returning any that are immutable.
+func (b *BulkLGTM) RegisterOptions(opts *options.Options) sets.String {
 	opts.RegisterInt(&b.maxDiff, "bulk-lgtm-max-diff", 10, "The maximum number of differences (additions + deletions) for PRs to include in the bulk LGTM list")
 	opts.RegisterInt(&b.maxChangedFiles, "bulk-lgtm-changed-files", 1, "The maximum number of changed files for PRs to include in the bulk LGTM list")
 	opts.RegisterInt(&b.maxCommits, "bulk-lgtm-max-commits", 1, "The maximum number of commits for PRs to include in the bulk LGTM list")
 	opts.RegisterDuration(&b.cookieDuration, "bulk-lgtm-cookie-duration", 24*time.Hour, "The duration for the cookie used to store github credentials.")
 	opts.RegisterBool(&b.disableSecureCookie, "bulk-lgtm-insecure-disable-secure-cookie", false, "If true, the cookie storing github credentials will be allowed on http")
-	opts.RegisterString(&githubOauthConfig.RedirectURL, "bulk-lgtm-github-oauth-redirect-url", "http://localhost:8080/bulkprs/callback", "The URL for the OAuth2 callback")
+	opts.RegisterString(&b.redirectURL, "bulk-lgtm-github-oauth-redirect-url", "http://localhost:8080/bulkprs/callback", "The URL for the OAuth2 callback")
```

Comment:
The githubOauthConfig struct is used by goroutines besides the main goroutine. If I bound the redirect flag directly to the githubOauthConfig.RedirectURL field there could be a data race on that field when options are updated.